manubot/cite/util.py

```diff
@@ -106,18 +112,41 @@ def is_valid_citation_string(string):
         return False
 
     if not source or not identifier:
-        msg = f'{string} → blank source or identifier'
+        msg = f'Invalid citation: {string}\nblank source or identifier'
         logging.error(msg)
         return False
 
-    # Exempted non-citation sources used for pandoc-fignos, pandoc-tablenos,
-    # and pandoc-eqnos
-    if source in {'fig', 'tbl', 'eq'}:
-        return False
+    if allow_pandoc_xnos:
+        # Exempted non-citation sources used for pandoc-fignos,
+        # pandoc-tablenos, and pandoc-eqnos
+        pandoc_xnos_keys = {'fig', 'tbl', 'eq'}
+        if source in pandoc_xnos_keys:
+            return False
```

Comment:
I may be misreading the logic here.  If `allow_pandoc_xnos` and `source in pandoc_xnos_keys`, shouldn't this return True?

Reply:
Got it, that behavior makes sense.